src/Support/InferExtensions/ModelExtension.php

```diff
@@ -72,22 +72,34 @@ public function getPropertyType(PropertyFetchEvent $event): ?Type
         throw new \LogicException('Should not happen');
     }
 
-    private function getAttributeTypeFromDbColumnType(string $columnType): AbstractType
+    /**
+     * MySQL/MariaDB decimal is mapped to a string by PDO.
+     * Floating point numbers and decimals are all mapped to strings when using the pgsql driver.
+     */
+    private function getAttributeTypeFromDbColumnType(?string $columnType, ?string $dbDriverName): AbstractType
     {
-        $type = Str::before($columnType, ' ');
-        $typeName = Str::before($type, '(');
-
-        // @todo Fix to native types
-        $attributeType = match ($typeName) {
-            'int', 'integer', 'bigint' => new IntegerType(),
-            'float', 'double', 'decimal' => new FloatType(),
-            'varchar', 'string', 'text', 'datetime' => new StringType(), // string, text - needed?
-            'tinyint', 'bool', 'boolean' => new BooleanType(), // bool, boolean - needed?
-            'json', 'array' => new ArrayType(),
-            default => new UnknownType("unimplemented DB column type [$type]"),
-        };
+        if ($columnType === null) {
+            return new UnknownType();
+        }
+
+        $typeName = str($columnType)
+            ->before(' ') // strip modifiers from a type name such as `bigint unsigned`
+            ->before('(') // strip the length from a type name such as `tinyint(4)`
+            ->toString();
+
+        if (in_array($typeName, ['int', 'integer', 'tinyint', 'smallint', 'mediumint', 'bigint'])) {
```

Comment:
Wondering if I need to specify `smallserial, serial, bigserial` for `pgsql`

Reply:
@kburton-dev If you know how these types are mapped to PHP types, go ahead :-)

Reply:
Actually, I tested this now, and since those `serial` types are not true types (just notational convenience in pgsql), they are  just `integers` under the hood